scripts/deploy.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+// DEPLOYMENT HAPPENS HERE - tread carefully
+
+const path = require('path');
+const NetlifyAPI = require('netlify');
+const ora = require('ora');
+const logSymbols = require('log-symbols');
+
+const {dirs} = require('./utils.js');
+
+const BUILD_DIR = path.join(__dirname, '..', 'dist');
+const client = new NetlifyAPI(process.env.NETLIFY_TOKEN);
+
+function createOrUpdate (extensionName) {
+  const siteName = `ctf-exts-${extensionName}`;
+  const customDomain = `${extensionName}.contentfulexts.com`;
```

Comment:
why do we need a site per extension again?

Reply:
To allow for breaking changes. This way whenever we have to/want to make a breaking change we update the `majorVersion` property in `extension.json` which will automatically create a new subdomain. The old subdomain won't be touched again by this repository and will work forever. This way we don't impact existing installations.

A good example of why braking changes are sometimes necessary are sidebar extensions. If we had any extensions using the old sidebar model in the Marketplace, now would be the time to update the `majorVersion` to use the new API.
 
In theory, Netlify does allow for incremental updates and maintaining previous files. But it's far outside their happy path and I'd rather not rely on it.

Reply:
Alternative that doesn't produce all those Netlify sites would be to just write versioned directories to an S3 bucket and put a Cloudfront distribution in front of it.

Reply:
That would work. Though I actually quite like the fully independent sites. I've checked with Jani before building this, and he's fine with ~30 extra sites in the main account. If it becomes a problem, we can always move them to their own Netlify account.

Bonus advantage, we can use Netlify functions for some extensions. I got my eyes on the Marketo one for example.